NitroxServer/GameLogic/Entities/EntitySimulator.cs

```diff
@@ -92,11 +91,14 @@ private List<Entity> AssignForCells(Player player, AbsoluteEntityCell[] added)
             foreach (AbsoluteEntityCell cell in added)
             {
                 List<Entity> entities = entityManager.GetEntities(cell);
-
                 assignedEntities.AddRange(
-                    entities.Where(entity => cell.Level <= entity.Level &&
-                                                ((entity.SpawnedByServer && serverSpawnedSimulationWhiteList.Contains(entity.TechType)) || !entity.SpawnedByServer) &&
-                                                simulationOwnershipData.TryToAcquire(entity.Id, player, DEFAULT_ENTITY_SIMULATION_LOCKTYPE)));                       
+                    entities.Where(entity =>
+                    {
+                        bool isSpawnedByServerAndWhitelisted = entity.SpawnedByServer && serverSpawnedSimulationWhiteList.Contains(entity.TechType);
+                        bool shouldBeOwnershipSimulated = isSpawnedByServerAndWhitelisted || !entity.SpawnedByServer;
+                        bool canAquireLockForPlayer = simulationOwnershipData.TryToAcquire(entity.Id, player, DEFAULT_ENTITY_SIMULATION_LOCKTYPE);
```

Comment:
This needs to be the last statement because we do not want to try to acquire a lock without checking to see if we should.